File: MuizzaApp1/MuizzaApp1/Models/EmotionalResponse.cs
namespace MuizzaApp1.Models
{
    public class EmotionalResponse
    {
        public string Affirmation { get; set; }
        public string Explanation { get; set; }
        public DateTime CachedAt { get; set; }

        public EmotionalResponse(string affirmation, string explanation)
        {
            Affirmation = affirmation;
            Explanation = explanation;
            CachedAt = DateTime.UtcNow;
        }

        // Parameterless constructor for deserialization
        public EmotionalResponse()
        {
        }
    }
} 

File: MuizzaApp/MuizzaApp/Models/EmotionalResponse.swift
import Foundation

struct EmotionalResponse: Codable {
    var affirmation: String
    var explanation: String
    var cachedAt: Date

    init(affirmation: String, explanation: String) {
        self.affirmation = affirmation
        self.explanation = explanation
        self.cachedAt = Date()
    }
}
